types/index.d.ts

```diff
@@ -0,0 +1,318 @@
+interface AuthOptions {
+  clientId: string;
+  persistentCache: boolean;
+  redirectUri: string;
+}
+
+interface AzureAuthOptions {
+  /** @param {String} clientId your AzureAuth client identifier */
+  clientId: string;
+  /** @param {boolean} [persistentCache] should store token cache between the app starts; defaults to true */
+  persistentCache?: string;
+  /** @param {String} [authorityUrl] optional Azure authority if you want to replace default v2 endpoint (`https://login.microsoftonline.com/${tenant}/oauth2/v2.0/`) */
+  authorityUrl?: string;
+  /** @param {String} [tenant] uses given tenant in the default authority URL; default is `common` */
+  tenant?: string;
+  /** @param {String} [redirectUri] - uses given redirect URI instead of default one:
+   *
+   *  iOS: {YOUR_BUNDLE_IDENTIFIER}://${YOUR_BUNDLE_IDENTIFIER}/ios/callback
+   *
+   *  Android: {YOUR_APP_PACKAGE_NAME}://{YOUR_APP_PACKAGE_NAME}/android/callback
+   */
+  redirectUri?: string;
+}
+
+interface ClientOptions {
+  token: string;
```

Comment:
Considering that https://github.com/vmurin/react-native-azure-auth/blob/v1.5.1/src/networking/index.js#L30 defines an empty object as default value for the options argument, it seems like all properties in this interface should be optional.

```suggestion
  token?: string;
```

Reply:
`token` is optional anyway